controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -1138,21 +1144,49 @@ int bml_internal::process_cmdu_header(std::shared_ptr<beerocks_header> beerocks_
                 return BML_RET_OP_FAILED;
             }
 
-            //Signal any waiting threads
-            if (m_prmClientListGet) {
-                bool promise_result = false;
-                if (m_client_list && m_client_list_size) {
-                    *m_client_list_size = *m_client_list_size <= response->client_list_size()
-                                              ? response->client_list_size()
-                                              : *m_client_list_size;
-                    *m_client_list = response->client_list_str();
-                    promise_result = true;
+            if (!m_client_list || !m_client_list_size) {
+                LOG(ERROR) << "The pointer to the user data buffer is null!";
+                m_prmClientListGet->set_value(response->result() == 0);
```

Comment:
Should the promise fail if we encounter an error?
```suggestion
                m_prmClientListGet->set_value(false);
```

Reply:
Also on the follow-up error cases.

Reply:
fixed - thanks.
this made me consider other locations as well.